src/OrbitClientGgp/ClientGgp.cpp

```diff
@@ -91,14 +95,28 @@ bool ClientGgp::RequestStartCapture(ThreadPool* thread_pool) {
   TracepointInfoSet selected_tracepoints;
   bool collect_thread_state = absl::GetFlag(FLAGS_thread_state);
   bool enable_introspection = false;
-  ErrorMessageOr<void> result = capture_client_->StartCapture(
+  Future<ErrorMessageOr<CaptureOutcome>> result = capture_client_->StartCapture(
       thread_pool, target_process_, module_manager_, selected_functions_, selected_tracepoints,
       absl::flat_hash_set<uint64_t>{}, collect_thread_state, enable_introspection);
 
-  if (result.has_error()) {
-    ERROR("Error starting capture: %s", result.error().message());
-    return false;
-  }
+  orbit_base::ImmediateExecutor executer;
+  result.Then(&executer, [this](ErrorMessageOr<CaptureOutcome> result) {
+    if (!result) {
+      ClearCapture();
+      ERROR("Capture failed: %s", result.error().message());
+      return;
+    }
+
+    switch (result.value()) {
```

Comment:
Nit/personal preference: if you decide to you use the "implicit bool conversion" above in favour of the `has_value`, I'd also use the `*` here in favour of the `value()`, i.e. `switch (*result) {`

As always for personal pref., feel totally free to ignore it.

Reply:
I changed is to use `has_value()`